### ISSUE TITLE
Doesn't work without empty settings

### ISSUE THREAD

**Issue by kocv59** [microsoft/vscode-eslint#2008]
The empty settings are needed to work. Without `useFlatConfig` or `nodePath` there's an error. `Request textDocument/diagnostic failed with message: The "path" argument must be of type string. Received undefined` or `Request textDocument/diagnostic failed with message: Cannot read properties of undefined (reading 'useFlatConfig')`.

Without `rulesCustomizations` and `problems` there's no error but eslint doesn't do anything.

```
nodePath = "",
experimental = {
	useFlatConfig = false,
},
problems = {},
rulesCustomizations = {},
```

**Comment by kocv59** [comment]
This link is not functioning for me?

**Comment by kocv59** [comment]
The client is Neovim, but that shouldn't be a concern. There should be suitable defaults in place so that no additional settings are necessary.

For instance, the flat config has been the default in ESLint for a long time, so it shouldn't generate an error if you don't specify it; `useFlatConfig = true` should be the default. The same applies to the other values mentioned in the issue; they should not require setting to an empty string or object. It should work seamlessly out of the box.

**Comment by dbaeumer** [comment]
@kocv59 as I was trying to say the server it currently tailored to run in VS Code. I have never made an effort to ensure that the server is running out of the box in other editors as well. There were efforts from the community to do so but none got pushed over the finish line. I am happy to help if someone is interested to give this another try.

**Comment by kocv59** [comment]
It's already working in other editors all I was saying is provide defaults for the settings I showed so people don't get confused when it breaks without empty settings. It's probably 5 changed lines to set the defaults.

**Comment by ataha322** [comment]
@kocv59 hey did you figure out what fields are missing in the settings?
